crd-generator/api/src/main/java/io/fabric8/crd/generator/AbstractJsonSchema.java

```diff
@@ -387,25 +389,20 @@ private T internalFromImpl(String name, TypeRef typeRef, Set<String> visited) {
       return arrayLikeProperty(schema);
     } else if (io.sundr.model.utils.Collections.IS_MAP.apply(typeRef)) { // Handle Maps
       final TypeRef keyType = TypeAs.UNWRAP_MAP_KEY_OF.apply(typeRef);
-      boolean degraded = false;
-      if (keyType instanceof ClassRef) {
-        ClassRef classRef = (ClassRef) keyType;
-        if (!classRef.getFullyQualifiedName().equals("java.lang.String")) {
-          degraded = true;
-        } else {
-          final TypeRef valueType = TypeAs.UNWRAP_MAP_VALUE_OF.apply(typeRef);
-          classRef = (ClassRef) valueType;
-          if (!classRef.getFullyQualifiedName().equals("java.lang.String")) {
-            degraded = true;
-          }
-        }
-      } else {
-        degraded = true;
+
+      if (!(keyType instanceof ClassRef && ((ClassRef) keyType).getFullyQualifiedName().equals("java.lang.String"))) {
+        LOGGER.warn("Property '{}' with '{}' key type is mapped to 'string' because of CRD schemas limitations", name, typeRef);
       }
-      if (degraded) {
-        LOGGER.warn("Property '{}' with '{}' type is mapped to a string to string mapping because of CRD schemas limitations", name, typeRef);
+
+      final TypeRef valueType = TypeAs.UNWRAP_MAP_VALUE_OF.apply(typeRef);
+      T schema = internalFromImpl(name, valueType, visited);
+
+      if (schema == null) {
+        LOGGER.warn("Property '{}' with '{}' value type is mapped to 'object' because it's CRD representation cannot be extracted.", name, typeRef);
```

Comment:
```suggestion
        LOGGER.warn("Property '{}' with '{}' value type is mapped to 'object' because its CRD representation cannot be extracted.", name, typeRef);
```